### PR TITLE
Scrapyd batch processing

### PR DESCRIPTION
**Pull request by samarthdd**
Implemented scrapyd

Implemented batch processing which will sync .scrapy cache to s3 bucket 

That .scrapy will be synched to s3 when scraper is stoped

It will be downloaded back from s3 when scraper starts to track files which are already downloaded

Fixed virustotal report issue

### CODE REVIEW

File: gw_crawler/malicious_file_crawler/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+RUN pip install python-dotenv==0.13.0
+RUN pip install scrapy-dotpersistence==0.3.0
+RUN pip install scrapyd
+RUN pip install git+https://github.com/scrapy/scrapyd-client
+RUN pip install python-scrapyd-api
+
+COPY . .
+
+CMD set -m; scrapyd & cd . && scrapyd-deploy  && fg scrapyd
+
```
Comment (susilupwork) [inline gw_crawler/malicious_file_crawler/Dockerfile:20]: Why so many blank  lines. Use PyCharm to take care of formatting.

File: compressed_file_generator/src/utils/minio_service.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
+import logging
+
+from minio import Minio
+from src.config import Config
+logger = logging.getLogger("GW:minio")
+
+
+class MinioService:
```
Comment (susilupwork) [inline compressed_file_generator/src/utils/minio_service.py:8]: Do not copy paste Minio_service or s3_client code for every microservice. Put it in gw.testdata.util  or some place. Use it as a library.  Or you need extra method update in library.

File: gw_crawler/malicious_file_crawler/src/run_distributed_crawl.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
+""" Main class for initializing and running crawlers. """
```
Comment (susilupwork) [inline gw_crawler/malicious_file_crawler/src/run_distributed_crawl.py:1]: Good comment done.

**Review by susilupwork — CHANGES_REQUESTED** [review]
Good comment done.
Do not copy paste Minio_service or s3_client code for every microservice. Put it in gw.testdata.util  or some place. Use it as a library.  Or you need extra method update in library.

File: gw_crawler/malicious_file_crawler/Dockerfile (hunk 1 above)
Comment (samarthdd) [reply gw_crawler/malicious_file_crawler/Dockerfile:20]: Removed extra spaces

File: compressed_file_generator/src/utils/minio_service.py (hunk 2 above)
Comment (samarthdd) [reply compressed_file_generator/src/utils/minio_service.py:8]: Address this in PR : 102

**Review by dtollaku — APPROVED** [review]

**Pull request merged**